pkg/workflows/wasm/host/module_test.go

```diff
@@ -48,14 +49,25 @@ func newMockMessageEmitter(e func(string, map[string]string) error) custmsg.Mess
 // access functions are injected as mocks.
 func Test_createEmitFn(t *testing.T) {
 	t.Run("success", func(t *testing.T) {
+		store := &store{
+			m:  make(map[string]*RequestData),
+			mu: sync.RWMutex{},
+		}
+		reqId := "random-id"
+		err := store.add(
+			reqId,
+			&RequestData{ctx: func() context.Context { return tests.Context(t) }})
+		require.NoError(t, err)
 		emitFn := createEmitFn(
 			logger.Test(t),
+			store,
 			newMockMessageEmitter(func(_ string, _ map[string]string) error {
 				return nil
 			}),
```

Comment:
I would modify the tests now to enable you to make an assertion on the context, add a key to the context and verify it can be read out in the message emitter

Reply:
```suggestion
			store,
			newMockMessageEmitter(func(ctx context.Context, _ string, _ map[string]string) error {
			         v := context.Value(ctx)
			         require.Equal(t, v, "test-value)
				return nil
			}),
```